content/docs/rfcs/73/README.md

```diff
@@ -0,0 +1,205 @@
+
+Note that `txid` may not always be practical as proof of payment due to on-chain confirmation latency.
+To address this issue, future versions of the protocol may involve:
+- paying for multiple requests in one transaction;
+- using faster (likely L2-based) payment mechanisms.
+
+## Wire Format Specifications for Store PoC incentivization
+
+### Request
+
+We extend `HistoryQuery` to include an eligibility proof:
+```protobuf
+message HistoryQuery {
+  // the first field is reserved for future use
+  string pubsubtopic = 2;
+  repeated ContentFilter contentFilters = 3;
+  PagingInfo pagingInfo = 4;
+  // numbering gap left for non-eligibility-related protocol extensions
+  + optional bytes eligibility_proof = 10;
+}
+```
+
+An example of usage with `txid` as a proof of payment:
+```protobuf
+HistoryQuery history_query {
+  // the first field is reserved for future use
+  pubsubtopic: "example_pubsub_topic"
+  contentFilters: []
+  pagingInfo: {
+    // provide values for PagingInfo fields
+  }
+  eligibility_proof: {
+	  proof_of_payment: 0xabc123  // txid for the client's payment
+	  // eligibility proofs of other types are not included
+  };
+}
+```
+
+### Response
+
+We extend the `HistoryResponse` to indicate the eligibility status:
+```protobuf
+message HistoryResponse {
+  // the first field is reserved for future use
+  repeated WakuMessage messages = 2;
+  PagingInfo pagingInfo = 3;
+  enum Error {
+    NONE = 0;
+    INVALID_CURSOR = 1;
+    + NON_ELIGIBLE = 2;
+  }
+  Error error = 4;
+  + EligibilityStatus eligibility_status = 5;
+}
+```
+
+Example of a response if the client is eligible:
+```protobuf
+HistoryResponse response_example {
+  messages: [message_1, message_2]
+  pagingInfo: paging_info
+  error: NONE
+  eligibility_status: {
+	  status_code: 200
+	  status_desc: "OK"
+  }
+}
+```
+
+Example of a response if the client is not eligible:
+```protobuf
+HistoryResponse response_example {
+  messages: [] // no messages sent to non-eligible clients
+  pagingInfo: paging_info
+  error: NON_ELIGIBLE
+  eligibility_status: {
+    status_code: 402
+    status_desc: "PAYMENT_REQUIRED"
+  }
+}
+```
+
+# Security/Privacy Considerations
+
+Eligibility proofs may reveal private information about the client.
+In particular, a transaction identifier used as a proof of payment links the client's query to their on-chain activity.
+Potential countermeasures may include using one-time addresses or ZK-based privacy-preserving protocols.
+
+# Copyright
+
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
+
+# References
+
+## normative
+- A high-level [incentivization outline](https://github.com/waku-org/research/blob/master/incentivization.md)
+- [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/) (for Store-specific sections)
+
+## informative
+
+RFCs of request-response protocols:
+- [12/WAKU2-FILTER](https://rfc.vac.dev/spec/12/)
```

Comment:
@s-tikhomirov, great specification! I just wanted to mention that all links to rfc.vac.dev should be displayed as "/spec/12"